bin/handlers/pnpm-handler.ts

```diff
@@ -29,6 +35,127 @@ import {
 // regex to detect options section in help text
 const OPTIONS_SECTION_RE = /^\s*Options:/i;
 
+function getPnpmStorePaths(): string[] {
+  const paths = ['~/.pnpm-store'];
+
+  if (directoryExists('.pnpm-store')) {
+    paths.push('./.pnpm-store');
+  }
+
+  return paths;
+}
+
+// completion handlers for pnpm options that take values
+const pnpmOptionHandlers = {
+  dir: function (complete: (value: string, description: string) => void) {
+    complete('./', 'Current directory');
+    complete('../', 'Parent directory');
+
+    const dirs = getDirectoriesInCwd();
+    for (const dir of dirs) {
+      if (dir !== './') {
+        complete(dir, `Directory: ${dir.slice(2)}`);
+      }
+    }
+  },
+
+  loglevel: function (complete: (value: string, description: string) => void) {
+    complete('debug', 'Debug level');
+    complete('info', 'Info level');
+    complete('warn', 'Warning level');
+    complete('error', 'Error level');
+    complete('silent', 'Silent level');
+  },
+
+  reporter: function (complete: (value: string, description: string) => void) {
+    complete('default', 'Default reporter');
+    complete('silent', 'Silent reporter');
+    complete('append-only', 'Append-only reporter');
+    complete('ndjson', 'NDJSON reporter');
+  },
+
+  filter: function (complete: (value: string, description: string) => void) {
+    complete('.', 'Current working directory');
+    complete('...', 'Include dependents');
+
+    const workspaceDirs = getCommonWorkspaceDirs();
+    for (const dir of workspaceDirs) {
+      complete(`${dir}/*`, `All packages in ${dir.slice(2)}`);
+    }
+
+    complete('@*/*', 'All scoped packages');
+    complete('@types/*', 'All type packages');
+  },
+
+  'modules-dir': function (
```

Comment:
i don't think we need autocompletions for this, the shell will handle it hopefully.